cmd/operator/main.go

```diff
@@ -152,6 +154,7 @@ func init() {
 	flagset.StringVar(&cfg.AlertManagerSelector, "alertmanager-instance-selector", "", "Label selector to filter AlertManager CRDs to manage")
 	flagset.Parse(os.Args[1:])
 	cfg.Namespaces = ns.asSlice()
+	cfg.NoNamespaces = noNs.asSlice()
```

Comment:
ok, this is the actual problem here, why e2e hangs. :sweat_smile: `asSlice()` returns a slice with one empty string entry `[]string{""}`, if `--no-namespaces` is not set. Hence, all empty string namespace events `""` will be denied.

In k8s, the default namespace is emitted as empty string, causing those events never to be propagated, effectively halting the system.

verdict: e2e was right, just not very communicative.